app/presenters/data_set_csv_presenter.rb

```diff
@@ -17,11 +17,48 @@ def to_csv
 
   def to_array_for_csv
     [].tap do |csv|
-      headers = ['name', 'address1', 'address2', 'town', 'postcode', 'access_notes', 'general_notes', 'url', 'lat', 'lng', 'phone', 'fax', 'text_phone']
-      csv << headers
+      csv << all_headers
       places.each do |place|
-        csv << headers.collect { |h| place.send(h.to_sym) }
+        csv << build_row(place)
       end
     end
   end
+
+private
+
+  # We need to handle the location fields separately from the others.
+  # location is a Point object (so doesn't serialize well to CSV by default)
+  # and doesn't need to be exported because it's either overridden from the
+  # override fields or set from MapIt's data using the postcode.
+  # Also exclude the id from the CSV since it isn't needed by the import.
+  def non_location_headers
+    Place.attribute_names - ['_id', 'location', 'override_lng', 'override_lat']
+  end
+
+  # The fields called 'lng' and 'lat' are used as the override values by
+  # Place.parameters_from_hash so we need to name them accordingly in the
+  # CSV to avoid them being lost if a CSV is exported and then imported.
+  def location_headers
+    ['lng', 'lat']
+  end
+
+  def all_headers
+    non_location_headers + location_headers
+  end
+
+  def build_row(place)
+    add_location_fields(place, non_location_fields(place))
+  end
+
+  def non_location_fields(place)
+    non_location_headers.map { |header| place.send(header.to_sym) }
+  end
+
+  def add_location_fields(place, row)
+    if place.override_lat_lng?
+      row + [place.override_lng, place.override_lat]
+    else
+      row + [nil, nil]
+    end
+  end
```

Comment:
How about `build_row` and `add_location_fields` become as follows:

```ruby
def build_row(place)
  non_location_fields(place) + location_fields(place)
end

def location_fields(place, row)
  if place.override_lat_lng?
    [place.override_lng, place.override_lat]
  else
    [nil, nil]
  end
end
```

Yes, it is fairly nitpicky but it would give us some parity between the `xxx_headers` and `xxx_fields` methods which would be more pleasing (to me).  Definitely not a blocker to merging though.